File: apps/identity/src/errors/error-code.ts
import { ErrorCode } from '../../../../protogen/ts/common/v1/error_pb.js';

export { ErrorCode };

export function toPublicErrorCode(errorCode: ErrorCode): string {
  const code = ErrorCode[errorCode];

  if (errorCode === ErrorCode.UNSPECIFIED || typeof code !== 'string') {
    return ErrorCode[ErrorCode.INTERNAL_ERROR] as string;
  }

  return code;
}

File: apps/payments/internal/errorcode/error_code.go
package errorcode

import (
	"strings"

	commonv1 "polyglot-ticketing-v2/protogen/go/common/v1"
)

const enumPrefix = "ERROR_CODE_"

func String(code commonv1.ErrorCode) string {
	name := code.String()
	if code == commonv1.ErrorCode_ERROR_CODE_UNSPECIFIED || !strings.HasPrefix(name, enumPrefix) {
		name = commonv1.ErrorCode_ERROR_CODE_INTERNAL_ERROR.String()
	}
	return strings.TrimPrefix(name, enumPrefix)
}
